include/path_tracking_pid/controller.hpp

```diff
@@ -98,11 +98,10 @@ class Controller : private boost::noncopyable
    * @return tf of found position on plan
    * @return index of current path-pose if requested
    */
-  tf2::Transform findPositionOnPlan(const geometry_msgs::Transform current_tf,
-                                    ControllerState* controller_state_ptr,
-                                    size_t &path_pose_idx);
+  tf2::Transform findPositionOnPlan(geometry_msgs::Transform current_tf, ControllerState* controller_state_ptr,
+                                    size_t& path_pose_idx);
```

Comment:
Shouldn't most of ROS message arguments be const references?
```c++
const geometry_msgs::Transform& current_tf
```

Reply:
Yeah, I think so. I checked `geometry_msgs::Transform` and it is quite large. Too large to pass by value, I'd say.

clang-tidy has a warning for inefficient argument passing and it fired on another function, so I assumed it would have checked everything. Apparently it is unable to do this for every function or it uses different heuristics. I will check all functions tomorrow.

Reply:
@Rayman fixed!